cmd/gardenlet/app/migration.go

```diff
@@ -6,10 +6,150 @@ package app
+		return fmt.Errorf("failed listing namespaces: %w", err)
+	}
+
+	var (
+		tasks []flow.TaskFn
+		crbs  = []string{v1beta1constants.ShootProjectAdminsGroupName, v1beta1constants.ShootProjectViewersGroupName, v1beta1constants.ShootSystemAdminsGroupName, v1beta1constants.ShootSystemViewersGroupName}
+	)
+
+	for _, namespace := range namespaceList.Items {
+		if namespace.DeletionTimestamp != nil || namespace.Status.Phase == corev1.NamespaceTerminating {
+			continue
+		}
+
+		tasks = append(tasks, func(ctx context.Context) error {
+			var (
+				shootSystemKey             = client.ObjectKey{Namespace: namespace.Name, Name: shootsystem.ManagedResourceName}
+				shootSystemManagedResource = &resourcesv1alpha1.ManagedResource{}
+
+				gardenerAccessKey             = client.ObjectKey{Namespace: namespace.Name, Name: gardeneraccess.ManagedResourceName}
+				gardenerAccessManagedResource = &resourcesv1alpha1.ManagedResource{}
+			)
+
+			// Get the shoot-core-system managed resource and check if it is already migrated.
+			if err := seedClient.Get(ctx, shootSystemKey, shootSystemManagedResource); err != nil {
+				if apierrors.IsNotFound(err) {
+					log.Info("Managed resource not found, skipping migration", "managedResource", shootSystemKey)
+					return nil
+				}
+				return fmt.Errorf("failed to get ManagedResource %q: %w", shootSystemKey, err)
+			}
+
+			if shootSystemManagedResource.DeletionTimestamp != nil {
+				return nil
+			}
+
+			shootSystemObjects, err := managedresources.GetObjects(ctx, seedClient, shootSystemManagedResource.Namespace, shootSystemManagedResource.Name)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					log.Info("Managed resource secret not found, skipping migration", "managedResource", shootSystemKey)
+					return nil
+				}
+				return fmt.Errorf("failed to get objects for ManagedResource %q: %w", shootSystemKey, err)
+			}
+
+			oldShootSystemObjectsCount := len(shootSystemObjects)
+			shootSystemObjects = slices.DeleteFunc(shootSystemObjects, func(obj client.Object) bool {
+				return slices.Contains(crbs, obj.GetName())
+			})
+
+			if oldShootSystemObjectsCount == len(shootSystemObjects) {
+				log.Info("ClusterRoleBindings for shoot/adminkubeconfig and shoot/viewerkubeconfig have already been migrated, skipping", "managedResource", shootSystemKey)
+				return nil
+			}
+
+			// Move the ClusterRoleBindings to the shoot-core-gardeneraccess managed resource firstly.
+			if err := seedClient.Get(ctx, gardenerAccessKey, gardenerAccessManagedResource); err != nil {
+				if apierrors.IsNotFound(err) {
+					log.Info("Managed resource not found, skipping migration", "managedResource", gardenerAccessKey)
+					return nil
+				}
+				return fmt.Errorf("failed to get ManagedResource %q: %w", gardenerAccessKey, err)
+			}
+
+			if gardenerAccessManagedResource.DeletionTimestamp != nil {
+				return nil
```

Comment:
We could add a log entry for this early exit:
```suggestion
				log.Info("Managed resource is in deletion, skipping migration", "managedResource", gardenerAccessKey)
				return nil
```